lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php

```diff
@@ -432,6 +432,12 @@ private function avoidReusingParentProcessConnection()
             $this->parentConnections[] = $this->_connection;
             $this->_connection = null;
             $this->pid = getmypid();
+
+            // Reset config host to avoid issue with multiple connections
+            if (!empty($this->_config['port'])) {
```

Comment:
Can we add a check here if the port is not already set with localhost then only perform this action:
```suggestion
            if (!empty($this->_config['port']) && strpos($this->_config['host'], ':') === false) {
```